capstone/src/main/webapp/docsScript.js

```diff
@@ -0,0 +1,107 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// API key from the Developer Console
+const API_KEY = 'API_KEY'; // TODO: Create Java servlet to return key
+
+// Array of API discovery doc URLs for APIs used by the quickstart
+const DISCOVERY_DOCS = ["https://docs.googleapis.com/$discovery/rest?version=v1"];
+
+// Authorization scopes required by the API; multiple scopes can be
+// included, separated by spaces.
+const SCOPES = "https://www.googleapis.com/auth/documents.readonly";
+
+/** Collects the user's input and submits it for analysis */
+async function gatherDocsInput() {
+  // Get the submitted id and language
+  const idElement = document.getElementById('doc-id');
+  if (!idElement) {
+    return;
+  }
+  let id = idElement.value;
+  const search = idElement.value.match(/\/d\/([\w-]+)/);
```

Comment:
It might be good to put a short comment for what this regular expression represents

Reply:
Got it!